src/Microsoft.Dnx.Tooling/Utils/LockFileUtils.cs

```diff
@@ -63,7 +66,59 @@ public static LockFileLibrary CreateLockFileLibrary(LockFileLibrary previousLibr
             return lockFileLib;
         }
 
-        public static LockFileTargetLibrary CreateLockFileTargetLibrary(LockFileLibrary library, IPackage package, RestoreContext context, string correctedPackageName)
+        public static LockFileLibrary CreateLockFileProjectLibrary(LockFileProjectLibrary previousLibrary,
+                                                                   Runtime.Project project,
+                                                                   Runtime.Project library)
+        {
+            var result = new LockFileProjectLibrary()
+            {
+                Name = library.Name,
+                Version = library.Version
+            };
+
+            if (previousLibrary?.Name == library.Name && previousLibrary?.Version == library.Version)
+            {
+                result.RelativePath = previousLibrary.RelativePath;
+            }
+            else
+            {
+                result.RelativePath = PathUtility.GetRelativePath(project.ProjectFilePath, library.ProjectFilePath, '/');
+            }
+
+            return result;
+        }
+
+        public static LockFileTargetLibrary CreateLockFileTargetLibrary(LockFileProjectLibrary library,
+                                                                        Runtime.Project projectInfo,
+                                                                        RestoreContext context)
+        {
+            var lockFileLib = new LockFileTargetLibrary
+            {
+                Name = library.Name,
+                Version = library.Version
+            };
+
+            var dependencies = (IEnumerable<Runtime.LibraryDependency>)projectInfo.Dependencies;
```

Comment:
This logic should match https://github.com/aspnet/dnx/blob/dev/src/Microsoft.Dnx.Runtime/DependencyManagement/ProjectReferenceDependencyProvider.cs#L47.